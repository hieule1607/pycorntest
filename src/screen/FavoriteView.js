import React, { Component } from 'react';
import { StyleSheet, View, Text, FlatList } from 'react-native';
import Database from '../Database';
import CustomImage from "../component/PycoImage";
const db = new Database();
export default class FavoriteView extends Component {

  state = {
    isLoading: false,
    listUser: []
  }

  static navigationOptions = ({navigation}) => {
    return {
      title: 'Favorite',      
    }
  };

  componentDidMount() {
    this.getProducts();
  }

  getProducts = () => {
    let listUser = [];
    db.getPycoUsers().then((data) => {
      listUser = data;
      this.setState({
        listUser,
        isLoading: false,
      });
    }).catch((err) => {
      console.log(err);
      this.setState = {
        isLoading: false
      }
    })
  }
  
  keyExtractor = (item, index) => index.toString()

  renderItem = ({ item }) => {
    const user = item.user
    return (
    <View>
      <CustomImage styleView={{width: 80, height: 80, margin: 10}} style={{margin: 4}} uri={user.picture}/>
    </View>    
    )
  }
  
  render() {
    if(this.state.isLoading){
      return(
        <View style={styles.activity}>
          <ActivityIndicator size="large" color="#0000ff"/>
        </View>
      )
    }
    if(this.state.listUser.length === 0){
      return(
        <View>
          <Text style={styles.message}>{this.state.notFound}</Text>
        </View>
      )
    }
    return (
      <FlatList
        contentContainerStyle={{alignSelf: 'center', alignItems: 'stretch'}}
        style={{padding: 8}}
        numColumns={3}
        keyExtractor={this.keyExtractor}
        data={this.state.listUser}
        renderItem={this.renderItem}
      />
    );
  }
}

const styles = StyleSheet.create({
  container: {
   flex: 1,
   paddingBottom: 22
  },
  item: {
    padding: 10,
    fontSize: 18,
    height: 44,
  },
  activity: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: 'center',
    justifyContent: 'center'
  },
  message: {
    padding: 16,
    fontSize: 18,
    color: 'red'
  }
});