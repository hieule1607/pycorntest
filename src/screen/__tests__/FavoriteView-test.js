import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import FavoriteView from "../FavoriteView";
import { FlatList } from 'react-native';

describe(`FavoriteView Component`, () => {
    let props
    let wrapper

    beforeEach(() => {
        wrapper = shallow(<FavoriteView {...props} />);
      });

    it(`should render correctly`, () => {
        const tree = renderer.create(<FavoriteView {...props}/>).toJSON()
        expect(tree).toMatchSnapshot();
    })

    it(`should rendered Flatlist`, () => {
        expect(wrapper.find(FlatList)).toMatchSnapshot();
    })

    it('should check `componentDidMount()`', () => {
        const instance = wrapper.instance();
        jest.spyOn(instance, 'getProducts');
        instance.componentDidMount();
        expect(instance.getProducts).toHaveBeenCalledTimes(1);
    });
})