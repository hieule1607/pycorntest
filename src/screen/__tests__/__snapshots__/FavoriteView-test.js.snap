// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteView Component should render correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "red",
        "fontSize": 18,
        "padding": 16,
      }
    }
  />
</View>
`;

exports[`FavoriteView Component should rendered Flatlist 1`] = `ShallowWrapper {}`;
